src/base/bittorrent/session.cpp

```diff
@@ -3816,22 +3817,30 @@ bool Session::addMoveTorrentStorageJob(TorrentHandleImpl *torrent, const QString
         if (iter != m_moveStorageQueue.end()) {
             // remove existing inactive job
             m_moveStorageQueue.erase(iter);
+            LogMsg(tr("Cancelled to move \"%1\" from \"%2\" to \"%3\".").arg(torrent->name(), currentLocation, iter->path));
         }
     }
 
-    QString currentLocation = torrent->actualStorageLocation();
     if (!m_moveStorageQueue.isEmpty() && (m_moveStorageQueue.first().torrentHandle == torrentHandle)) {
-        // if there is active job for this torrent consider its target path as current location
-        // of this torrent to prevent creating meaningless job that will do nothing
-        currentLocation = m_moveStorageQueue.first().path;
+        // if there is active job for this torrent prevent creating meaningless
+        // job that will move torrent to the same location as current one
+        if (QDir {m_moveStorageQueue.first().path} == QDir {newPath}) {
+            LogMsg(tr("Couldn't enqueue move of \"%1\" to \"%2\". Torrent is currently moving to the same destination location.")
+                   .arg(torrent->name(), newPath));
+            return false;
+        }
+    }
+    else {
+        if (QDir {currentLocation} == QDir {newPath}) {
+            LogMsg(tr("Couldn't enqueue move of \"%1\" from \"%2\" to \"%3\". Both paths point to the same location.")
```

Comment:
And we could omit source path in this message, it is the same as the destination path and the message is clear about it.

Reply:
As I mentioned, there may be several different paths pointing to the same place. Can't it be useful for the user to see both of them here? From my point of view, this is the only message where the source path has any meaning.

P.S. I don't want to stop there, so if you want me to change some messages, specify it explicitly (without any hesitation) and let's get it over with.